### PR TITLE
Make fixtures files full-fledged members of manifest and enable partial parsing

### LINKED ISSUE #9067: [CT-3380] [Implementation] Include file-based fixtures in parsed files during partial parsing
### Housekeeping

- [X] I am a maintainer of dbt-core

### Short description

If a file-based unit test fixture is modified between dbt invocations, this should be detected in partial parsing and trigger the scheduling of parsing UnitTestDefinition nodes that depend on the fixture. 

Otherwise, it's possible to make changes to a fixture file that won't be detected in state:modified when running unit tests. 

### Acceptance criteria

* remove `--no-partial-parse` flag from this test: TODO

### Impact to Other Teams

N/A

### Will backports be required?

nope - working off a net-new feature branch. 

### Context

Initial discussion: https://github.com/dbt-labs/dbt-core/pull/9032

### PR DESCRIPTION
**Pull request by gshank**
resolves #9067

### Problem

Because fixture files were not full-fledged SourceFiles and didn't have a representation in the manifest, we couldn't do partial parsing for fixture file changes.

### Solution

Create a new source file class, FixtureSourceFile, and a lightweight UnitTestFixture node-like object in the manifest fixtures dictionary. Store the unique_ids of the fixtures and the related unit_tests in the source_file and process them in partial parsing.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

File: core/dbt/graph/selector_methods.py [hunk 1]
```diff
@@ -727,21 +727,21 @@ def search(self, included_nodes: Set[UniqueId], selector: str) -> Iterator[Uniqu
 
         manifest: WritableManifest = self.previous_state.manifest
 
-        for node, real_node in self.all_nodes(included_nodes):
+        for unique_id, node in self.all_nodes(included_nodes):
```
Comment (MichelleArk) [inline core/dbt/graph/selector_methods.py:730]: thank you for renaming this here -- this has been bugging me forever!

**Comment by MichelleArk** [comment]
This [TODO in test_state](https://github.com/dbt-labs/dbt-core/pull/9032/files#diff-0f0c14a88170fc0b4f9e86ec195ef3df5a15801b97ccca70dfcefe1c58210832R73)can also be cleaned up as part of this work! It should test the appropriate tests are selected when a file fixture is updated.

**Review by MichelleArk — COMMENTED** [review]
Looking great! No blocking concerns :)

File: core/dbt/parser/unit_tests.py [hunk 2]
```diff
@@ -272,6 +275,33 @@ def parse(self) -> ParseResult:
...
+                fixture_source_file.unit_tests.append(unit_test_definition.unique_id)
+        if unit_test_definition.expect.fixture:
+            # find fixture file object and store unit_test_definition unique_id
+            fixture_source_file = self.get_fixture_source_file(
+                unit_test_definition.expect.fixture,
+                self.project.project_name,
+            )
+            fixture_source_file.unit_tests.append(unit_test_definition.unique_id)
+
+    def get_fixture_source_file(self, fixture_name: str, project_name: str):
```
[the remark this answers is not in the document]
Comment (gshank) [reply core/dbt/parser/unit_tests.py]: Done

**Pull request merged**